src/codegen/object.rs

```diff
@@ -340,6 +333,27 @@ pub fn generate(
     Ok(())
 }
 
+fn generate_default(w: &mut dyn Write, env: &Env, analysis: &analysis::object::Info) -> Result<()> {
+    let name = &analysis.name;
+    if let Some(func) = analysis.default_constructor() {
+        general::declare_default(w, env, name, func)
+    } else if has_builder_properties(&analysis.builder_properties) {
```

Comment:
The reason why with this patch RadioButton etc get picked up is because before we were falling back to object::new only if there was a `new()` function, but it did not have 0 params.

That seems rather odd and with the new code I did not get any false positive in core and gtk3